src/main/java/org/mule/extension/aggregator/internal/config/AggregatorManager.java

```diff
@@ -188,6 +188,24 @@ public void registerListener(String aggregatorName, AggregatorListener listener)
     registeredListeners.put(aggregatorName, listener);
   }
 
+  /**
+   * Unregisters a listener from an already registered aggregator
+   *
+   * @param aggregatorName the name of the aggregator to unregister from
+   * @param listener the listener to be called when needed
+   * @throws MuleRuntimeException
+   */
+  public void unregisterListener(String aggregatorName, AggregatorListener listener) throws MuleRuntimeException {
+    if (!availableAggregators.containsKey(aggregatorName)) {
+      throw new MuleRuntimeException(createStaticMessage("Listener is attempting to unregister from aggregator: '%s', but it does not exist",
+                                                         aggregatorName));
+    }
+    boolean unregisterSuccessful = registeredListeners.remove(aggregatorName, listener);
```

Comment:
value returned by registeredListener.remove can be used directly in the if to avoid that extra var.

Reply:
I know, but for the sake of readability, I decided to make it explicit. But, yes, I think it doesn't hurt to do the method call and the condition check at the same time because it's a simple expression in this case.